libs/libapi/tests/test_duckdb.py

```diff
@@ -0,0 +1,58 @@
+from pathlib import Path
+from typing import Optional
+from unittest.mock import patch
+
+import pytest
+from pytest import TempPathFactory
+
+from libapi.duckdb import get_index_file_location_and_download_if_missing
+
+
+@pytest.mark.parametrize("partial_index", [False, True])
+@pytest.mark.parametrize("partial_split", [False, True])
+def test_get_index_file_location_and_download_if_missing(
+    partial_split: bool, partial_index: bool, tmp_path_factory: TempPathFactory
+) -> None:
+    duckdb_index_file_directory = (
+        tmp_path_factory.mktemp("test_get_index_file_location_and_download_if_missing") / "duckdb"
+    )
+    duckdb_index_file_directory.mkdir(parents=True, exist_ok=True)
+
+    dataset = "dataset"
+    revision = "revision"
+    target_revision = "refs/convert/parquet"
+    config = "config"
+    split = "split"
+    default_filename = "index.duckdb"
+    split_directory = f"partial-{split}" if partial_split else split
+    filename = f"partial-{default_filename}" if partial_index else default_filename
```

Comment:
OH ok... So we can have:

`train/index.duckdb`: normal
`train/partial-index.duckdb`: the dataset is originally in Parquet format and contains more than 5GB data: the Parquet "conversion" is complete, while the duckdb index is only on the first 5GB
`partial-train/partial-index.duckdb`: if the Parquet files have been created only on the first 5GB of the data
`partial-train/index.duckdb`: ... we should never have this, right?

Reply:
`partial-train/index.duckdb` we can have this if the index contains all the data from the partial parquet export

Reply:
OK. Hence, is it `partial-train/partial-index.duckdb` that will never appear?

Reply:
well, it should not occur, but... it could, if the parquet files have been created on the first 5GB of data, and the result is N parquet files, where at most N-1 parquet files sum at least 5GB. I don't think it's possible, but... OK